libbeat/processors/add_host_metadata/config.go

```diff
@@ -34,10 +35,24 @@ type Config struct {
 }
 
 func defaultConfig() Config {
-	return Config{
-		NetInfoEnabled:      true,
-		CacheTTL:            5 * time.Minute,
-		ExpireUpdateTimeout: time.Second * 10,
-		ReplaceFields:       true,
+	// Setting environmental variable ELASTIC_NETINFO:false in Elastic Agent pod will disable the netinfo.enabled option of add_host_metadata processor
+	// This will result to events not being enhanced with host.ip and host.mac
+	// Related to https://github.com/elastic/integrations/issues/6674
+	valueNETINFO, _ := os.LookupEnv("ELASTIC_NETINFO")
+
+	if valueNETINFO == "false" {
+		return Config{
+			NetInfoEnabled:      false,
+			CacheTTL:            5 * time.Minute,
+			ExpireUpdateTimeout: time.Second * 10,
+			ReplaceFields:       true,
+		}
+	} else {
+		return Config{
+			NetInfoEnabled:      true,
+			CacheTTL:            5 * time.Minute,
+			ExpireUpdateTimeout: time.Second * 10,
+			ReplaceFields:       true,
+		}
```

Comment:
Optional suggestion to increase readability

```suggestion
	return Config{
		NetInfoEnabled:      valueNETINFO == "true",
		CacheTTL:            5 * time.Minute,
		ExpireUpdateTimeout: time.Second * 10,
		ReplaceFields:       true,
	}
```